core/common/src/MapLayerSettings.ts

```diff
@@ -311,6 +329,11 @@ export class ImageMapLayerSettings extends MapLayerSettings {
     if (props.queryParams) {
       this.savedQueryParams = {...props.queryParams};
     }
+
+    if (props.properties) {
+      this.properties = {...props.properties}
```

Comment:
Note: this does not deep-copy arrays, if you are using spread operator due to concerns about mutating the input.

Reply:
good catch, what about `structuredClone` ?

Reply:
If you're truly concerned, sure. Or just document that people shouldn't modify the input after passing it to you.